### PR TITLE
SASS-2298 Create What type of reduced allowance page

### PR DESCRIPTION
**Pull request by HarryGull**
### Description
Create 'What type of reduced annual allowance do you have? page

Add a link to the relevant story in Jira
[SASS-SASS-2298](https://jira.tools.tax.service.gov.uk/browse/SASS-2298)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+        )
+
+        insertCyaData(pensionsUserDataWithAnnualAllowances(pensionsViewModel), aUserRequest)
+        urlPost(fullUrl(reducedAnnualAllowanceTypeUrl(taxYearEOY)), body = validFormAllChecked, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        //TODO - go to annual allowance cya page when available
```
Comment (ILee15) [inline it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala:614]: TODO should be "Do you have a reduced annual allowance?" page

File: it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
```
Comment (ILee15) [inline it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala:590]: TODO should be "Do you have a reduced annual allowance?" page

File: it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+        )
+
+        insertCyaData(pensionsUserDataWithAnnualAllowances(pensionsViewModel), aUserRequest)
+        urlPost(fullUrl(reducedAnnualAllowanceTypeUrl(taxYearEOY)), body = form, follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has a SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        // navigate to 'Have you gone above your reduced annual allowance?' page when available
```
Comment (ILee15) [inline it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala:529]: Could you put TODO in this comment here

File: it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+        )
+
+        insertCyaData(pensionsUserDataWithAnnualAllowances(pensionsViewModel), aUserRequest)
+        urlGet(fullUrl(reducedAnnualAllowanceTypeUrl(taxYearEOY)), follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        //TODO - go to annual allowance cya page when available
```
Comment (ILee15) [inline it/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeControllersControllerISpec.scala:366]: TODO should be "Do you have a reduced annual allowance?" page

same for the next test as well

File: app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+              pensionSessionService.createOrUpdateSessionData(request.user,
+                updatedCyaModel, taxYear, data.isPriorSubmission)(errorHandler.internalServerError()) {
+                // navigate to 'Have you gone above your reduced annual allowance?' page when available
+                Redirect(PensionsSummaryController.show(taxYear))
+              }
+            }
+          )
+        } else {
+          // navigate to annual allowance CYA when available
+          Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
```
Comment (ILee15) [inline app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala:86]: Should this one also redirect to ""Have you gone above the annual allowance?" question page" e.g. if it's been answered false or is None

File: app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+      case Some(data) =>
+        if (data.pensions.pensionsAnnualAllowances.reducedAnnualAllowanceQuestion.contains(true)) {
+          val taperedAnnualAllowance = data.pensions.pensionsAnnualAllowances.taperedAnnualAllowance
+          val moneyPurchaseAnnualAllowance = data.pensions.pensionsAnnualAllowances.moneyPurchaseAnnualAllowance
+          val form = ReducedAnnualAllowanceTypeQuestionForm.reducedAnnualAllowanceTypeForm(request.user.isAgent)
+          Future.successful(Ok(view(form, taxYear, moneyPurchaseAnnualAllowance, taperedAnnualAllowance)))
+        } else {
+          // navigate to annual allowance CYA when available
```
[the remark this answers is not in the document]
Comment (HarryGull) [reply app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala]: This should go to cya if no session, or the question page if question is false (if question is  true it is processed)
I have applied this distinction in the redirect

[the remark this answers is not in the document]
Comment (HarryGull) [reply app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala]: This should go to the question page 
so changed to
//TODO: navigate to 'Do you have a reduced annual allowance?' page when available

File: app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala (hunk 5 above)
Comment (HarryGull) [reply app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala:86]: yes
But it is the missing question page 'have you gone above the annual allowance page 'rather then the next question" suggested

So changed to:
//TODO: navigate to 'Do you have a reduced annual allowance?' page when available

**Review by ILee15 — APPROVED** [review]

**Pull request merged**